sdklab/regressions/regression_github_990.py

```diff
@@ -0,0 +1,284 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import asyncio
+import logging
+import functools
+import ssl
+from test_utils import test_env, logging_hook
+from azure.iot.device.aio import IoTHubDeviceClient
+import azure.iot.device.common
+
+
+logging.basicConfig(level=logging.WARNING)
+logging.getLogger("azure.iot").setLevel(level=logging.DEBUG)
+
+"""
+This app simulates was a bug that was reported where a websockets connection was raising a
+`TlsExchangeAuthError` on a reconnect, between `PUBLISH` and `PUBACK`.
+
+Order of events for this bug repro:
+1. Customer calls `send_message`
+2. After `PUBLISH` and before `PUBACK`, transport disconnects with `rc=1`
+3. On reconnnect, `transport.connect` raised `TlsExchangeAuthError(None,)` caused by SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)')`
+4. `TlsExchangeAuthError` is not transient, so `reconnect` fails. Since the reconnect is not user-initiated, no error is returned to the caller.
+5. `PUBACK` never received, so `send_message` never completes.
+
+The fix is to make `TlsExchangeAuthError` into a retryable error, so a reconnect gets initiated after step 3 above.
+
+In the interest of completeness, this test was expanded to 8 different scenarios based on 3 flags
+
+* Half of them simulate a retryable error (`TlsExchangeAuthError`), and half of them simulate a non-retryable error (bare `Exception`).
```

Comment:
I would prefer if you used an exception you defined instead, since bare Exception carries some risk. Significantly less while not in a pytest context, but the same issue can arise anytime you use bare Excepts - there's always the potential that something is catching it and acting unexpectedly as a result.